src/init/starship.ps1

```diff
@@ -136,22 +136,37 @@ $null = New-Module starship {
         $arguments += "--status=$($lastExitCodeForPrompt)"
 
         # Invoke Starship
-        $promptText = if ($script:TransientPrompt) {
+        $promptText, $rpromptText = $null, $null
+        if ($script:TransientPrompt) {
             $script:TransientPrompt = $false
-            if (Test-Path function:Invoke-Starship-TransientFunction) {
+            $promptText = if (Test-Path function:Invoke-Starship-TransientFunction) {
                 Invoke-Starship-TransientFunction
             } else {
                 "$([char]0x1B)[1;32m❯$([char]0x1B)[0m "
             }
+            $rpromptText = if (Test-Path function:Invoke-Starship-TransientRightFunction) {
+                Invoke-Starship-TransientRightFunction
+            } else {
+                ""
+            }
         } else {
-            Invoke-Native -Executable ::STARSHIP:: -Arguments $arguments
+            $promptText = Invoke-Native -Executable ::STARSHIP:: -Arguments $arguments
+            $arguments += "--right"
+            $rpromptText = Invoke-Native -Executable ::STARSHIP:: -Arguments $arguments
         }
 
+        $promptLines = $promptText.Split("`n")
+        # Calculate offset for printing on right side:
+        # offset = (console width) + (length of ANSI sequences on right_prompt) - (length of just text in last line of left_prompt)
+        $rpromptOffset = $Host.UI.RawUI.WindowSize.Width + ($rpromptText.Length - ($rpromptText -replace $ansiRegex).Length) - ($promptLines[-1] -replace $ansiRegex).Length
+
         # Set the number of extra lines in the prompt for PSReadLine prompt redraw.
-        Set-PSReadLineOption -ExtraPromptLineCount ($promptText.Split("`n").Length - 1)
+        Set-PSReadLineOption -ExtraPromptLineCount ($promptLines.Length - 1)
 
         # Return the prompt
-        $promptText
+        # After printing the left prompt, the cursor position is saved,
+        # right prompt is printed, and then cursor position is restored.
+        "$promptText$([char]0x1B)[s{0,$rpromptOffset}$([char]0x1B)[u" -f $rpromptText
```

Comment:
Do you know if there is a notable difference between this and the ANSI escape I used?

Please also avoid printing the rprompt if it wouldn't fit like my PR did.

Reply:
> Do you know if there is a notable difference between this and the ANSI escape I used?

Is there any documentation on `1B7` and `1B8`? I couldn't figure out how to use it in PowerShell.

Reply:
https://en.wikipedia.org/wiki/ANSI_escape_code#Fp_Escape_sequences It does seem to be former private use, so it might be worse.

Reply:
I guess for clarity and familiarity, the `[s` and `[u` would be better, since all the Google search results for cursor control show these sequences.

Reply:
> Please also avoid printing the rprompt if it wouldn't fit like my PR did.

Done